src/lib.rs

```diff
@@ -0,0 +1,350 @@
+extern crate core;
+
+pub mod commands;
+
+use crate::commands::CargoCommand;
+use cargo_metadata::{Message, MetadataCommand};
+use core::fmt;
+use rustc_version::Channel;
+use semver::Version;
+use serde::Deserialize;
+use std::io::{BufRead, BufReader};
+use std::path::{Path, PathBuf};
+use std::process::{Command, ExitStatus, Stdio};
+use std::{env, io, process};
+use tee::TeeReader;
+
+pub fn build_elf(
+    cmd: CargoCommand,
+    message_format: &str,
+    args: &Vec<String>,
+) -> (ExitStatus, Vec<Message>) {
+    let mut command = make_cargo_build_command(cmd, message_format, args);
+    let mut process = command.spawn().unwrap();
+    let command_stdout = process.stdout.take().unwrap();
+
+    let mut tee_reader;
+    let mut stdout_reader;
+    let buf_reader: &mut dyn BufRead = if message_format == "json-render-diagnostics" {
```

Comment:
"json-render-diagnostics" used to be behind a constant. Let's change back to using a constant.